src/main/java/org/mockito/internal/runners/util/RunnerProvider.java

```diff
@@ -4,24 +4,24 @@
  */
 package org.mockito.internal.runners.util;
 
+import org.mockito.internal.runners.RunnerImpl;
+
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
 
-import org.mockito.internal.runners.RunnerImpl;
-
 public class RunnerProvider {
 
-    public RunnerImpl newInstance(String runnerClassName, Class<?> constructorParam) throws Exception {
+    public RunnerImpl newInstance(String runnerClassName, Object ... constructorArgs) throws Exception {
         Constructor<?> constructor;
         try {
             Class<?> runnerClass = Class.forName(runnerClassName);
-            constructor = runnerClass.getConstructor(Class.class.getClass());
+            constructor = runnerClass.getConstructors()[0];
```

Comment:
It's not safe to rely on order of constructor, I don't believe the JVM garantees it. If there's a single constructor that's ok, but any implementor can add another (e.g. for tests), etc..